srcmedia/ts/members-map.ts

```diff
@@ -0,0 +1,127 @@
+import { map, icon, marker, tileLayer, control, featureGroup, PopupEvent } from 'leaflet'
+import { TiledMapLayer } from 'esri-leaflet'
+
+// map data - defined in the global scope outside this module, in the template
+type Address = {
+    name: string,
+    street_address: string,
+    city: string,
+    postal_code: number,
+    latitude: number,
+    longitude: number,
+    start_date: string,
+    end_date: string,
+}
+
+declare const address_data: Array<Address>
+
+/*
+ * map object with custom zoom control in bottom right
+ */
+const target = document.getElementById('address-map') as HTMLDivElement
+const addressMap = map(target, { zoomControl: false })
+const zoomControl = control.zoom({ position: 'bottomright' })
+
+zoomControl.addTo(addressMap)
+
+/*
+* basic tiled basemap from mapbox
+*/
+const mapboxToken = 'pk.eyJ1IjoicHJpbmNldG9uLWNkaCIsImEiOiJjazJrd2lybnEwMHdsM2JvM3UyMHUwbm02In0.4GJpwErZHkJH1DU-E-72OA'
+const basemap = tileLayer(`https://api.mapbox.com/styles/v1/mapbox/light-v10/tiles/{z}/{x}/{y}?access_token=${mapboxToken}`, {
+    attribution: 'Tiles <a href="https://apps.mapbox.com/feedback/">Mapbox</a>, <a href="http://www.openstreetmap.org/copyright">OpenStreetMap</a>'
+})
+
+basemap.addTo(addressMap)
+
+/*
+ * historic map of paris overlay
+ */
+const parisTiles = new TiledMapLayer({
+    url: 'https://tiles.arcgis.com/tiles/4Ko8f1mCWFLyY4NV/arcgis/rest/services/Paris_1943/MapServer',
+    attribution: '<a href="https://maps.princeton.edu/catalog/princeton-2r36tz994">Princeton University Library</a>',
+})
+
+parisTiles.addTo(addressMap)
+
+/*
+ * bookstore icon + marker on map
+ */
+const bookstoreIcon = icon({
```

Comment:
Can the bookstore info also be in a config? Doesn't seem like it should be in the map code

Reply:
thinking about this more - is the way to do it actually to create a `Address` in the database (maybe associated with Beach??) to store this info, and then pass it in the template? not sure, but it seems a little weird to store as a setting when it's actually sort of representing a model that we have in the db